src/utils/get-repo-data.js

```diff
@@ -8,63 +8,56 @@ const gitconfiglocal = require('gitconfiglocal')
 const isEmpty = require('lodash/isEmpty')
 const parseGitRemote = require('parse-github-url')
 
-const getRepoData = async function (remote) {
-  const cwd = process.cwd()
-  let repo = {}
+const getRepoData = async function ({ log, remoteName }) {
   try {
-    const gitConfig = await util.promisify(gitconfiglocal)(cwd)
-    const gitDirectory = findUp.sync(['.git'], { cwd, type: 'directory' })
-    const baseGitPath = path.dirname(gitDirectory)
+    const cwd = process.cwd()
+    const [gitConfig, gitDirectory] = await Promise.all([
+      util.promisify(gitconfiglocal)(cwd),
+      findUp(['.git'], { cwd, type: 'directory' }),
+    ])
+    const baseGitPath = dirname(gitDirectory)
 
     if (cwd !== baseGitPath) {
-      console.log(`Git directory located in ${baseGitPath}`)
-      // TODO prompt for "is this the correct git remote"?
-      // If folder gitignored inside another git repo it could link to wrong repo.
+      log(`Git directory located in ${baseGitPath}`)
     }
 
     if (isEmpty(gitConfig) || isEmpty(gitConfig.remote)) {
       throw new Error('No Git remote found')
     }
 
-    if (!remote) {
-      remote = Object.prototype.hasOwnProperty.call(gitConfig, 'origin')
```

Comment:
This is a very elaborate way to get the `origin` remote or the first one if `origin` doesn't exist.
Also there is a bug here `Object.prototype.hasOwnProperty.call(gitConfig, 'origin')` should be `Object.prototype.hasOwnProperty.call(gitConfig.remote, 'origin')`